omnipaxos/src/sequence_paxos/follower.rs

```diff
@@ -144,15 +134,38 @@ where
                         ],
                         "storage error while trying to write snapshot",
                     );
-                    let accepted_idx = self.internal_storage.append_entries(accsync.suffix);
+                    let accepted_res = self
+                        .internal_storage
+                        .append_entries_without_batching(accsync.suffix);
+                    // manually rollback set_snapshot if append suffix fails
+                    if let Err(_e) = &accepted_res {
+                        self.internal_storage
+                            .try_trim(self.internal_storage.get_accepted_idx())
+                            .expect("storage error while trying to trim log entries");
+                        self.internal_storage
+                            .append_entries_without_batching(old_log)
+                            .expect("storage error while trying to rollback log entries");
+                        if let Some(old_snapshot) = old_snapshot {
+                            self.internal_storage
+                                .set_snapshot(old_compacted_idx, old_snapshot)
+                                .expect("storage error while trying to rollback snapshot");
+                        } else {
+                            self.internal_storage
+                                .set_compacted_idx(old_compacted_idx)
+                                .expect("storage error while trying to rollback compacted index");
+                            self.internal_storage
+                                .reset_snapshot()
+                                .expect("storage error while trying to reset snapshot");
+                        }
```

Comment:
Same here, add a new function called `rollback_snapshot(s: Option<(Snapshot, u64)>` in InternalStorage that does this.